src/Object/AbstractPerson.php

```diff
@@ -270,15 +269,15 @@ public function getDateOfBirth()
     }
 
     /**
-     * @param \DateTime | \DateTimeInterface $dateOfBirth
+     * @param \DateTime|\DateTimeInterface|string $dateOfBirth
      */
     public function setDateOfBirth($dateOfBirth)
     {
         if ($dateOfBirth instanceof \DateTime
             || (is_object($dateOfBirth) && method_exists($dateOfBirth, 'format'))) {
             $this->dateOfBirth = $dateOfBirth->format('Y-m-d');
         } else {
-            throw new NostoException('Invalid argument, expected DateTime or DateTimeInterface');
+            $this->dateOfBirth = $dateOfBirth;
```

Comment:
I don't get this.

Reply:
It's because the setter and getter are inconsistent. Setter takes `\DateTimeInterface` but the getter always returns a string. The automatic encoder (https://github.com/Nosto/nosto-php-sdk/blob/feature/add-html-encoding/src/Mixins/HtmlEncoderTrait.php#L121) only works if the getter and setter are consistent.